### PR TITLE
Modifiying test_target_enter_exit_data_devices.c to use headerfile

### PR DESCRIPTION
**Pull request by tmh97**
Making style changes, adding macros, passes on clang and GCC.

### CODE REVIEW

**Comment by tmh97** [comment]
is target data map on line 72 redundant?

File: tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c [hunk 1]
```diff
@@ -17,29 +34,23 @@ int test_set_default_dev() {
   printf("initial device: %d\n", omp_get_initial_device());
   printf("default device: %d\n", def_dev);
```
Comment (josemonsalve2) [inline tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c]: Change all these printf's to OMPVV_INFOMSG

Comment (josemonsalve2) [inline tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c]: Change these to `OMPVV_INFOMSG`

File: tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c [hunk 2]
```diff
@@ -17,29 +34,23 @@ int test_set_default_dev() {
...
     omp_set_default_device(dev);
-    // unstructured mapping
-    {
+
 #pragma omp target enter data map(alloc: h_matrix[dev][0 : N])
-        printf(""); // forcing the compiler to not moving out of the scope
-    }
-    // operation
+    
     // assert(dev == omp_get_default_device() && "message");
```
Comment (josemonsalve2) [inline tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c]: remove this comment

**Comment by josemonsalve2** [comment]
I did not see the map on line 72. 


However, this test suffers from the same problem that @jhdavis8 's test has. If we decide not to use the TEST_SHARED_ENV option, it is not possible to make sure this test is compliant. But if we don't use this scheme we have no way we can fully guarantee that target regions are running on different devices

File: tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c [hunk 3]
```diff
@@ -1,45 +1,55 @@
...
 
-  printf("test_set_default_dev\n");
+  OMPVV_INFOMSG("test_set_default_dev");
 
   // Get number of devices
   int num_dev = omp_get_num_devices();
-  printf("num_devices: %d\n", num_dev);
+  OMPVV_INFOMSG("num_devices: %d", num_dev);
 
   int def_dev = omp_get_default_device();
```
Comment (spophale) [inline tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c]: Although I don't find a relationship between num_devices and default device no, I wonder if the device no is between 1 and num_dev.

Comment (tmh97) [reply tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c]: Check what value default is ->

Comment (tmh97) [reply tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c]: On Fatnode -> num_dev = 2, default_dev = 0, and initial device = -10

Comment (spophale) [reply tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c]: how are you getting initial device ?

Comment (tmh97) [reply tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c]: omp_get_initial_device()

**Pull request merged**